### PR TITLE
Milestone1 data exploration

### PR DESCRIPTION
**Pull request by tica1575**
Completed the initial data exploration for milestone 1

### CODE REVIEW

File: dev_sandbox/data_exploration_1D.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+rel_data_path = "data/X001_droplet_amplitudes.csv" # make sure you are in the very top directory of the project! 
+dframe = pd.read_csv(rel_data_path, header=0)
+print(dframe.head()) # take a look at the data
```
Comment (sstadick) [inline dev_sandbox/data_exploration_1D.py]: suggestion: Python style guide suggests 2 spaces before a comment when inline with code.

File: dev_sandbox/DataExploration_1D.ipynb [hunk 2]
```diff
@@ -0,0 +1,287 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "f00cf5bf",
+   "metadata": {},
+   "source": [
+    "# Milestone 1: Data Exploration\n",
+    "### Install Hints\n",
+    "Make sure to install your conda environment using the doc/envs/pydrop_dev.yml file for the most consistent results\n",
```
Comment (sstadick) [inline dev_sandbox/DataExploration_1D.ipynb:10]: suggestion(blocking): add a code block comment with the conda invocation to set up an env from that file.

File: dev_sandbox/DataExploration_1D.ipynb [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    "### Goals\n",
```
Comment (sstadick) [inline dev_sandbox/DataExploration_1D.ipynb:11]: praise: love the high level overview!

File: dev_sandbox/DataExploration_1D.ipynb [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "7bdae44b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
```
Comment (sstadick) [inline dev_sandbox/DataExploration_1D.ipynb:25]: suggestion(non-blocking): it's not required, but it's tidier to alphabetize your imports.

**Review by sstadick — APPROVED** [review]
@tica1575 : This is excellent, and is exactly what we were hoping for with some initial data exploration, excellent use of git as well!

File: dev_sandbox/DataExploration_1D.ipynb [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    }
+   ],
+   "source": [
+    "# to verify our mean and stdev. let's try fitting a basic normal distribution to see how skewed the data is:\n",
+    "bins = np.linspace(0, 6000, 100)\n",
+    "fig2 = plt.figure(2, figsize=[10,5])\n",
+    "ax1 = fig2.add_subplot(121)\n",
+    "ax2 = fig2.add_subplot(122)\n",
+    "bins = np.linspace(np.min(cluster0), np.max(cluster0), 100)\n",
+    "ax1.hist(cluster0, bins, label = 'cluster 0', color='r', density=True, alpha=0.5)\n",
```
Comment (stromk) [inline dev_sandbox/DataExploration_1D.ipynb]: nit: no space around assignments in function calls
```suggestion
    "ax1.hist(cluster0, bins, label='cluster 0', color='r', density=True, alpha=0.5)\n",
```

File: dev_sandbox/DataExploration_1D.ipynb [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+    "clustering = KMeans(n_clusters=K)\n",
+    "clustering.fit(X)\n",
+    "predictions = clustering.predict(X)\n",
+    "centers = clustering.cluster_centers_\n",
+    "if centers[0,0] < centers[1,0]: # flip so that cluster \"0\" corresponds to the \n",
+    "    predictions = np.logical_not(predictions) # cluster with lowest mean value\n",
+    "\n",
+    "overlap = predictions == true_y\n",
+    "true_overlap = overlap[overlap == 0]\n",
+    "frac_correct = len(true_overlap)/len(overlap)\n",
```
Comment (stromk) [inline dev_sandbox/DataExploration_1D.ipynb]: nit: space around operators
```suggestion
    "frac_correct = len(true_overlap) / len(overlap)\n",
```

**Review by stromk — CHANGES_REQUESTED** [review]
This looks great!

One more question to think about and address:
Looking at the graph you can see that there's a spread of data points between the two clusters. Can you figure out how many points lie in this middle area? If you had to make a metric for this, how would you define it? Give it a name and definition. In what ways would this metric be useful?

**Comment by sstadick** [comment]
https://conventionalcomments.org/ <- this is the style we're using for PR comments. In general it helps to remove ambiguity in code reviews. Since an off the cuff comment could cause someone to go off the rails and spend way too much time on a thing that really wasn't that important to the reviewer.

File: dev_sandbox/data_exploration_1D.py (hunk 1 above)
Comment (connordavel) [reply dev_sandbox/data_exploration_1D.py]: Should be edited now.

File: dev_sandbox/DataExploration_1D.ipynb (hunk 2 above)
Comment (connordavel) [reply dev_sandbox/DataExploration_1D.ipynb:10]: For this, could I refer to the README.md in the repo? I could say like "see ../README.md" for more info.

**Comment by ghost** [comment]
> This looks great!
> 
> One more question to think about and address: Looking at the graph you can see that there's a spread of data points between the two clusters. Can you figure out how many points lie in this middle area? If you had to make a metric for this, how would you define it? Give it a name and definition. In what ways would this metric be useful?

I put together some cursory code to address this. In this upcoming week I'm planning on expanding this to other distributions an analyzing and compiling the results into a Jupyter notebook. The name and definition we've given it are the interstitial fraction, and once refined and tested, this metric may be very helpful in preprocessing data before we undertake our clustering analysis.

**Review by stromk — APPROVED** [review]

**Pull request merged**